common/test/acceptance/pages/lms/edxnotes.py

```diff
@@ -538,3 +538,46 @@ def text(self, value):
         Sets text for the note.
         """
         self.q(css=self._bounded_selector(".annotator-item textarea")).first.fill(value)
+
+    @property
+    def tags(self):
+        """
+        Returns the tags associated with the note.
+
+        Tags are returned as a list of strings, with each tag as an individual string.
+        """
+        tag_text = []
+        self.show()
+        tags = self.q(css=self._bounded_selector(".annotator-annotation > div.annotator-tags > span.annotator-tag"))
+        if tags:
+            for tag in tags:
+                tag_text.append(tag.text)
+        self.q(css="body").first.click()
+        self.wait_for_notes_invisibility()
+        return tag_text
+
+    @tags.setter
+    def tags(self, tags):
+        """
+        Sets tags for the note. Tags should be supplied as a list of strings, with each tag as an individual string.
+        """
+        self.q(css=self._bounded_selector(".annotator-item input")).first.fill(" ".join(tags))
+
+    def has_sr_label(self, index, expected_text):
```

Comment:
@andy-armstrong agreed.  I would love to add support for testing the integrity of ```for -> id``` and ```aria-describedby -> id``` references across the platform.  Until then, any HTML and/or accessibility validator will catch such errors.  Since most of the dev work we do is behind HTTPauth, it makes it hard to run validation tools.  However, I recommend installing [Chrome's Accessibility Dev Tools](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb), which performs checks on the current DOM right in the inspector (Audits -> Accessibility -> Run).

There is a handy "Accessibility Properties" panel as well that exposes the role, state and properties, as well as what is spoken by a screen reader for any element in the DOM. 

This [Chromium ticket](https://code.google.com/p/chromium/issues/detail?id=390254#c25) indicates that they are soon adding native support for this tool in Chrome.  I have recently [requested a feature](https://code.google.com/p/chromium/issues/detail?id=390254#c25) that will synchronize tab focus with active node focus in the inspector which will make accessibility testing much, much easier.